From 1f5f5e09c155e9d3f9273aae429116b56d16bfee Mon Sep 17 00:00:00 2001
From: Catalin Popescu <catalin.popescu@intel.com>
Date: Tue, 8 Sep 2015 11:08:49 +0200
Subject: [PATCH] PMIC : fix out of bound accesses in irqregmap array.

Make sure irq_num is the size of irqregmap array.

Change-Id: Ief6debf9d593b35038a702ba80a8cac36ff86095
Signed-off-by: Catalin Popescu <catalin.popescu@intel.com>
---
 drivers/mfd/intel_soc_pmic_crc.c   | 5 ++---
 drivers/mfd/intel_soc_pmic_dc.c    | 3 ++-
 drivers/mfd/intel_soc_pmic_dc_ti.c | 3 ++-
 drivers/mfd/intel_soc_pmic_wc.c    | 5 ++---
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/mfd/intel_soc_pmic_crc.c b/drivers/mfd/intel_soc_pmic_crc.c
index 6c09518e1e1a..9111d4118880 100644
--- a/drivers/mfd/intel_soc_pmic_crc.c
+++ b/drivers/mfd/intel_soc_pmic_crc.c
@@ -28,8 +28,6 @@
 #include <linux/mfd/intel_soc_pmic.h>
 #include "intel_soc_pmic_core.h"
 
-#define CRYSTAL_COVE_IRQ_NUM	7
-
 #define CHIPID		0x00
 #define CHIPVER		0x01
 #define IRQLVL1		0x02
@@ -181,7 +179,8 @@ struct intel_soc_pmic crystal_cove_pmic = {
 	.init		= crystal_cove_init,
 	.cell_dev	= crystal_cove_dev,
 	.irq_regmap	= crystal_cove_irqregmap,
-	.irq_num	= CRYSTAL_COVE_IRQ_NUM,
+	.irq_num	= sizeof(crystal_cove_irqregmap)/
+				sizeof(crystal_cove_irqregmap[0]),
 };
 
 MODULE_LICENSE("GPL");
diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index a61788136c0f..25839e904b8e 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -679,7 +679,8 @@ struct intel_soc_pmic dollar_cove_pmic = {
 	.init		= dollar_cove_init,
 	.cell_dev       = dollar_cove_dev,
 	.irq_regmap	= dollar_cove_irqregmap,
-	.irq_num	= 48,
+	.irq_num	= sizeof(dollar_cove_irqregmap)/
+				sizeof(dollar_cove_irqregmap[0]),
 };
 
 MODULE_LICENSE("GPL");
diff --git a/drivers/mfd/intel_soc_pmic_dc_ti.c b/drivers/mfd/intel_soc_pmic_dc_ti.c
index 924b9ec035ab..827032988eec 100644
--- a/drivers/mfd/intel_soc_pmic_dc_ti.c
+++ b/drivers/mfd/intel_soc_pmic_dc_ti.c
@@ -163,7 +163,8 @@ struct intel_soc_pmic dollar_cove_ti_pmic = {
 	.init		= dollar_cove_ti_init,
 	.cell_dev	= dollar_cove_ti_dev,
 	.irq_regmap	= dollar_cove_ti_irqregmap,
-	.irq_num	= 8,
+	.irq_num	= sizeof(dollar_cove_ti_irqregmap)/
+				sizeof(dollar_cove_ti_irqregmap[0]),
 };
 
 MODULE_LICENSE("GPL");
diff --git a/drivers/mfd/intel_soc_pmic_wc.c b/drivers/mfd/intel_soc_pmic_wc.c
index dd6faab0bbea..681909502b30 100644
--- a/drivers/mfd/intel_soc_pmic_wc.c
+++ b/drivers/mfd/intel_soc_pmic_wc.c
@@ -48,8 +48,6 @@
 #include <linux/platform_device.h>
 
 
-#define WHISKEY_COVE_IRQ_NUM	17
-
 #define CHIPID		0x00
 #define CHIPVER	0x01
 
@@ -939,7 +937,8 @@ struct intel_soc_pmic whiskey_cove_pmic = {
 	.init		= whiskey_cove_init,
 	.cell_dev	= whiskey_cove_dev,
 	.irq_regmap	= whiskey_cove_irqregmap,
-	.irq_num	= WHISKEY_COVE_IRQ_NUM,
+	.irq_num	= sizeof(whiskey_cove_irqregmap)/
+				sizeof(whiskey_cove_irqregmap[0]),
 };
 
 #define TT_I2CDADDR_ADDR		0x00
-- 
2.6.0.rc0.2.g7662973

